hs/src/Reach/Eval/Core.hs

```diff
@@ -4124,14 +4128,45 @@ evalPrim p sargs =
     SLPrim_Contract_new -> do
       at <- withAt id
       let lab = "new Contract"
-      (ccv, opts) <-
+      let expectContractCode = \case
+            SLV_ContractCode _ cc -> return cc
+            ow -> expect_t ow $ Err_Expected "ContractCode or Reach.App"
+      (cc, opts) <-
         case args of
-          [x] -> return (x, mempty)
-          [x, y] -> (,) x <$> mustBeObject y
+          [x@(SLV_Prim (SLPrim_App_Delay _ _ _ _ mname))] -> do
+            -- Compile Reach.App and remember result of compilation
```

Comment:
Does the box get filled elsewhere?

Reply:
This comment is stale; I'll remove it. The box only needs to be written in `Compiler::m_compile_prog`, this way if you compile a file with

```JavaScript
export const child = Reach.App(() => {});

export const main = Reach.App(() => {
  // ...
  const ctc = new Contract(child);
});
```

The compiler will compile `child` first (alphabetical order), remember the result since the box is written in `Compiler.hs` and will not recompile inside of `new Contract`.